### DISCUSSION TITLE
[New] Eviter le préchargement des vidéos

### DISCUSSION THREAD

**Discussion by DocRoms** [cnumr/best-practices#46] [Bonnes pratiques] [answered]
### Bonne pratique : Eviter le préchargement des vidéos
Lorsque l'on charge une vidéo avec la balise `<video>` il est possible de ne pas activer l'autoplay, en n'utilisant pas l'attribut `autoplay`,  ce qui est déjà bien : 

```html
<!-- Avec Autoplay -->
<video src="fichiervideo.webm" autoplay></video>
<!-- Sans Autoplay -->
<video src="fichiervideo.webm"></video>
```

Cependant, même en désactivant l'autoplay, certaines partie de la vidéo, (métadatas ou plus), peuvent être téléchargéés. Pour éviter de télécharger des données qui ne seront jamais utilisées (si l'utilisateur ne lance pas la lecture de la vidéo), il est possible de forcer le navigateur de ne strictement rien précharger, en définissant la valeur de l'attribut `preload` de la balise `video` à `none` : 

```html
<!-- Avec Autoplay, mais la vidéo peut se charger en partie -->
<video src="fichiervideo.webm" autoplay></video>
<!-- Sans Autoplay mais la vidéo peut se charger en partie -->
<video src="fichiervideo.webm"></video>
<!-- Sans Autoplay et sans chargement de la vidéo -->
<video src="fichiervideo.webm" preload="none" ></video>
```

### Règle de validation proposée : 
    - Nombre de balise `<vidéo>` sans attribut `preload="none"` : 0

**Comment by ACTLEM** [comment, marked as answer]
https://github.com/cnumr/best-practices/issues/135

**Comment by Christophe-Cl** [comment]
Suggestion : 
- élargir à "vidéos, sons et animations interactives"
- ajouter la notion de : "laisser le user décider d'enclencher l'action de lire les contenus vidéos, sons et animations" (formulation à revoir). L'utilisateur doit avoir la maitrise sur les contenus.
